### PR TITLE
feat: Support public directory

### PR DESCRIPTION
**Pull request by carvinlo**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [ ] I have read the **CONTRIBUTING** document
- [ ] My code follows the code style of this project
- [ ] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

Use it in App Shell:
```html
<img src={`./img.png`} alt="" />
```
Use it in Pilet:
```html
<img src={`${__webpack_public_path__}img1.png`} alt="" />
```

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

File: src/tooling/piral-cli-webpack/src/set-path.ts [hunk 1]
```diff
@@ -11,7 +11,5 @@ function computePath() {
   return '/';
 }
 
-declare let __webpack_public_path__: string;
```
Comment (carvinlo) [inline src/tooling/piral-cli-webpack/src/set-path.ts:14]: Use `@types/webpack-env` instead.

**Comment by FlorianRappl** [comment]
Well, instead of 

```
<img src={`./img.png`} alt="" />
```

you'd use

```
<img src={require(`./img.png`)} alt="" />
```

anyway (as in all bundlers).

The other problem is that the public path is not really configurable (usually, you'd do this via your webpack config - i.e., the user would actually do the config you just did here).

The third problem is that you changed the injector. In my opinion this change is unnecessary. There is no need to distinguish between the cases. Also the public folder is copied / located at the destination, which makes this unnecessary.

Using the `webpack-env` is okay, even though for just a single variable I feel the change is unnecessary (which is why we did not include the package - its just one more download for nothing; as the used variable will never change in the future).

In general our point of view with Webpack is that we provide a configuration that does a couple things, but not as many as Parcel does out the box. This is also where Webpack and Parcel are different. The former requires (or wants) configuration, while the latter just works (but then is limited regarding configuration possibilities).

**Comment by carvinlo** [comment]
It would be even better if it could support the public directory, and the code might be more concise when using dynamic image paths. Because it is not processed by webpack, the build speed may be a little faster. The most important thing is that some developers such as I have become accustomed to using the public directory function. My point of view may have deviated. In any case, I understand, respect and support your ideas.

**Comment by FlorianRappl** [comment]
Yeah fully understood and appreciated. I agree that a folder for static files makes sense, however, usually you'd not want things like images in there, which you reference anyway in your code. You'd want things like `favicon.ico` etc. in there, which should be copied *without* a code reference.

But then again its unclear where the public folder should be. Some people like `src/public`, some `src/static`, some just `www` or `src/www` or `src/wwwroot`. You get the idea. It should be configurable. And then we can already omit the plugin directly and just refer to standard Webpack practices for introducing it.

**Comment by carvinlo** [comment]
Well, you are right, the directory name needs to be configurable. Is there any way to access the static resource directory in debug mode without modifying the injector? Am I right?

**Comment by FlorianRappl** [comment]
Why do you need to know the static resource directory? If you use the copy webpack plugin then all files should be copied to the resource directory. As such all requests are properly resolved. Maybe I am missing something. Can you elaborate why you think it should not work?

**Comment by carvinlo** [comment]
Oh, I’m very sorry, but I don’t know the copy webpack plugin thoroughly. In debug mode, public will be synchronized to the dist directory in real time:
image

So just add copy webpack plugin without any changes to piral, thanks for your help!

**Pull request closed without merging**